### DISCUSSION TITLE
Art direction: allow multiple crop variants for one image

### DISCUSSION THREAD

**Discussion by ursbraem** [craftcms/cms#8956] [Ideas]
Use case: An image has different aspect ratios for different breakpoints, and/or locations, e.g.: News image for 

* teaser on mobile (4:3)
* teaser on desktop (3:4)
* detail view on all devices (3:4)
* gallery view on all devices (unconstrained)

With the concept of "crop variants", it would be possible to allow client to make multiple, named crops for the same image in one place (so you upload the image for the news article and make all 3 required crops in one place)

Currently this is not possible, you can either use focal point or add multiple image fields to an article.

An approach is  that craft's image editor does not save one image, but all crop variants as pre-transform stage.

Configuring the image editor in `config/general.php`:

```
'imageEditorRatios' => [
  '3:4' => 0.75,
  '4:3' => 1.333,
  'Unconstrained' => 'none',
  'Original' => 'original',
],
```
Now the new behaviour: 

When you edit the example image in the image editor, and supposing the calculated aspect ratio is used as key, you get the following files:

* image-0.75.jpg
* image-1.333.jpg
* image-none.jpg
* image-original.jpg (or image.jpg)

This only regards aspect ratio. Scaling can be done in the template, as the client's manual contribution is not needed.

So to pick the 3:4 cropped  variant and transform it further in the template you can do (pseudo-code):

```
{% set thumb = {
    mode: 'scale',
    height: 100,
    quality: 75,
    imageEditorRatio: 0.75
} %}
```

If you want to base your transform on the original, do

```
{% set thumb = {
    mode: 'scale',
    height: 100,
    quality: 75,
    imageEditorRatio: 'original'
} %}
```

[related to #7117 (solved)]
[discussed with Oli]
[terminology: client=person in charge of editing, editor=craft's image editor]

**Comment by RitterKnightCreative** [comment]
IMO focus points get you 90% there but that other 10% you have to crop differently. 

I wonder if multiple focus points could help here?

Basically you set one which is the default focus point but you could also set another which you could then set which transform it should apply to, basically overriding the default crop.  When you move the focus point around, it changes the crop. 

Also see https://github.com/craftcms/cms/issues/1906

**Comment by piotrpog** [comment]
I think it would be pretty useful if we define multiple "variants" per asset source. Each could have different focal point and crop settings per asset in asset field.
OR we could decide to keep only one focal point, but multiple crop variants (useful for mobile where we might want to crop image more square-like than on desktop when it might be more panoramic).
